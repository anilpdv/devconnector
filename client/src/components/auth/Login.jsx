import React, { Component } from "react";
import { connect } from "react-redux";
import { loginUser } from "../../actions/auth";
import PropTypes from "prop-types";

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: ""
    };
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(e) {
    e.preventDefault();
    const userData = {
      email: this.state.email,
      password: this.state.password
    };
    this.props.loginUser(userData);
  }

  componentDidUpdate() {
    if (this.props.auth) {
      if (this.props.auth.isAuthenticated === true) {
        this.props.history.push("/dashboard");
      }
    }
  }

  render() {
    const { errors } = this.props;

    return (
      <div className="login">
        <div className="container">
          <div className="row">
            <div className="col-md-8 m-auto">
              {errors ? (
                errors.errors ? (
                  <div className="alert alert-danger text-center" role="alert">
                    {errors.errors ? errors.errors[0].msg.capitalize() : ""}
                  </div>
                ) : (
                  ""
                )
              ) : (
                ""
              )}
              <h1 className="display-4 text-center">Log In</h1>
              <p className="lead text-center">
                Sign in to your DevConnector account
              </p>
              <form action="dashboard.html" onSubmit={this.onSubmit}>
                <div className="form-group">
                  <input
                    type="email"
                    className="form-control form-control-lg email"
                    placeholder="Email Address"
                    name="email"
                    value={this.state.email}
                    onChange={e => this.setState({ email: e.target.value })}
                  />
                </div>
                <div className="form-group">
                  <input
                    type="password"
                    className="form-control form-control-lg password"
                    placeholder="Password"
                    name="password"
                    value={this.state.password}
                    onChange={e => this.setState({ password: e.target.value })}
                  />
                </div>
                <input type="submit" className="btn btn-info btn-block mt-4" />
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  errors: PropTypes.object.isRequired,
  auth: PropTypes.object.isRequired
};

const mapStateToProps = state => {
  return {
    errors: state.errors,
    auth: state.auth
  };
};

export default connect(
  mapStateToProps,
  { loginUser }
)(Login);
