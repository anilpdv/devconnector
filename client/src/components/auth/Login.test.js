import React from "react";
import { shallow, mount } from "enzyme";
import { Login } from "./Login";

describe("Login Component", () => {
  const login = mount(<Login />);
  it("should render without crashing", () => {
    expect(login).toMatchSnapshot();
  });

  describe("when the user input in to form", () => {
    beforeEach(() => {
      login.find(".email").simulate("change", {
        target: {
          value: "pdvanil007@gmail.com"
        }
      });
      login.find(".password").simulate("change", {
        target: {
          value: "password"
        }
      });
    });

    it("should match the email", () => {
      console.log(login.state());
      expect(login.state().email).toEqual("pdvanil007@gmail.com");
    });

    it("should match the password", () => {
      expect(login.state().password).toEqual("password");
    });
  });
});

describe("Login ", () => {
  const mockfn = jest.fn();
  const props = { errors: {}, loginUser: mockfn };
  const login = shallow(<Login {...props} />);

  beforeEach(() => {
    login.find(".email").simulate("change", {
      target: {
        value: "pdvanil007@gmail.com"
      }
    });
    login.find(".password").simulate("change", {
      target: {
        value: "password"
      }
    });

    login.find("form").simulate("submit", {
      preventDefault: () => {}
    });
  });

  it("should call the function with user data", () => {
    expect(mockfn).toHaveBeenCalledWith({
      email: "pdvanil007@gmail.com",
      password: "password"
    });
  });
});
