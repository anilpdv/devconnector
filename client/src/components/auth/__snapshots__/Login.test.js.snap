// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render without crashing 1`] = `ReactWrapper {}`;
